todo-app/src/Stores/TasksListStore.js

```diff
@@ -0,0 +1,151 @@
+        description: '',
+        date: '',
+    }
+
+    @action
+    selected = (id) => {
+        const todoItems = this.todoList.map(task => {
+            if (task.id === id) {
+                return {
+                    "id": task.id,
+                    "description": task.description,
+                    "deadline": task.deadline,
+                    "selected": !task.selected,
+                    "isDone": task.isDone,
+                    "timeout": task.timeout
+                }
+            } else {
+                return task
+            }
+        })
+        this.todoList = todoItems
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    setDone = (id) => {
+        const todoItems = this.todoList.map(task => {
+            if (task.id === id) {
+                return {
+                    "id": task.id,
+                    "description": task.description,
+                    "deadline": task.deadline,
+                    "selected": task.selected,
+                    "isDone": !task.isDone,
+                    "timeout": task.timeout
+                }
+            } else {
+                return task
+            }
+        })
+        this.todoList = todoItems
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    addNewTask = (newTask, newDeadLine) => {
+
+        const todoItems = this.todoList
+        const id = this.todoList.map(task => task.id)
+        const newId = this.todoList.length > 0 ? Math.max(...id) + 1 : 1
+        todoItems.push({
+            "id": newId,
+            "description": newTask.toUpperCase(),
+            "deadline": newDeadLine,
+            "selected": false,
+            "isDone": false,
+            "timeout": this.setTimeout(newDeadLine)
+        })
+        this.todoList = todoItems
+        this.newTask.description = ''
+        this.newTask.date = ''
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    setNewTask = (event, field) => {
+        this.newTask[field] = event.target.value
+    }
+
+    @action
+    setTimeout = (deadline) => {
+        let taskDeadline = moment(deadline).toObject()
+        if (moment([taskDeadline.years, taskDeadline.months, taskDeadline.date]).diff(moment(), 'days') < 0) {
+            console.log('true')
+            return true
+        } else {
+            console.log('false')
+            return false
+         }
+
+    }
+
+    @action
+    deleteTask = (id) => {
+
+        const todoItems = this.todoList.filter(task => task.id !== id)
+        this.todoList = todoItems
+        saveToLocalStorage(this.todoList)
+    }
+
+    @action
+    selectToEdit = (id) => {
+        this.selectedToEdit = id
+        this.editedTask = ''
+    }
+
+    @action
+    updateTask = (id, update) => {
+        const todoItems = this.todoList.map(task => {
+            if (task.id === id) {
+                return {
+                    "id": task.id,
+                    "description": update,

```

Comment:
You seem to be using these full objects each time, even when updating a single field. You should use either the spread operator, or `Object.assign`. Here's an example of spreading:

    return {
      ...task,
      description: update,
    }